sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/AadCredentialTest.java

```diff
@@ -53,20 +54,37 @@ private void setup(HttpClient httpClient, ConfigurationServiceVersion serviceVer
                 .httpClient(httpClient)
                 .credential(tokenCredential)
                 .endpoint(endpoint)
-                .serviceVersion(serviceVersion);
-
+                .serviceVersion(serviceVersion)
+                .httpLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS));
+            builder = setHttpClient(httpClient, builder);
             if (interceptorManager.isRecordMode()) {
-                builder.addPolicy(interceptorManager.getRecordPolicy()); // Record
+                builder.addPolicy(interceptorManager.getRecordPolicy())
+                    .addPolicy(new RetryPolicy()); // Record
             }
             client = builder.buildClient();
         }
     }
 
-    @Disabled("Disable it until resource is in production. The resource for testing is still a dogfood environment.")
+    ConfigurationClientBuilder setHttpClient(HttpClient httpClient, ConfigurationClientBuilder builder) {
+        if (interceptorManager.isRecordMode()) {
+            return builder
+                .httpClient(buildSyncAssertingClient(httpClient));
```

Comment:
Don't we want to add the asserting client when running LIVE test mode as well?

Reply:
Yea. It should also apply to LIVE mode as well. Updated.